docs/ref/models/querysets.txt

```diff
@@ -268,6 +268,46 @@ control the name of the annotation::
 For an in-depth discussion of aggregation, see :doc:`the topic guide on
 Aggregation </topics/db/aggregation>`.
 
+``alias()``
+~~~~~~~~~~~
+
+.. method:: alias(*args, **kwargs)
+
+.. versionadded:: 3.2
+
+Same as :meth:`~django.db.models.query.QuerySet.annotate()`, but instead of
+annotating objects in the ``QuerySet``, saves the expression for later reuse with
+other ``QuerySet`` methods. This is useful when
+the result of the expression itself is not needed but is used for filtering,
+ordering, or as a part of a complex expression. Not selecting the unused value
+removes redundant work from the database which should result in better performance.
+
+For example, if you were manipulating a list of blogs, you may want
+to find blogs with more than 5 entries, but not actually interested
+in the exact number of entries::
+
+    >>> from django.db.models import Count
+    >>> blogs = Blog.objects.alias(entries=Count('entry')).filter(entries__gt=5)
+
+``alias()`` can be used in conjunction with
+:meth:`~django.db.models.query.QuerySet.annotate()`,
+:meth:`~django.db.models.query.QuerySet.filter()`,
+:meth:`~django.db.models.query.QuerySet.order_by()`
+and :meth:`~django.db.models.query.QuerySet.update()`.
+To use aliased expression with other methods (e.g. aggregate), one must promote alias to annotation:
+
+    >>> Blog.objects.alias(entries=Count('entry')).annotate(entries=F('entries')).aggregate(Sum('entries'))
+
+:meth:`~django.db.models.query.QuerySet.filter()` and
+:meth:`~django.db.models.query.QuerySet.order_by()`
+can take expressions directly, but expression construction and usage often does
+not happen in the same place (for example, ``QuerySet``
+method creates expression, to be later used by the view).
+Also ``alias()`` allows you to build complex expressions incrementally,
+possibly spanning multiple methods and modules, refer to the expression parts
+by their aliases and only use
+:meth:`~django.db.models.query.QuerySet.annotate()` for the final result.
```

Comment:
Wrap docs at 79 chars. You can also simplify all references to `QuerySet`\'s methods in this doc, e.g. `` :meth:`~django.db.models.query.QuerySet.filter()` `` -> `` :meth:`filter` ``. Also `()` are added automatically when you use `:meth:`.